File: lab_2/input/input3.ts
interface Person {
  name: string;
  age: number;
  isStudent: boolean;
}

interface Book {
  title: string;
  pages: number;
}

function greet(name: string): void {
  console.log(`Hello, ${name}!`);
}

function add(x: number, y: number): number {
  const result = x + y;
  return result;
}

function createPerson(name: string, age: number, isStudent: boolean): Person {
  return {
      name: name,
      age: age,
      isStudent: isStudent
  };
}

function updateBook(book: Book, newTitle: string): Book {
  const updated = { title: newTitle, pages: book.pages };
  console.log(`Book updated to: ${newTitle}`);
  return updated;
}

const isAdult = (person: Person): boolean => {
  return person.age >= 18;
};

const numbers = [1, 2, 3, 4, 5];

const users = [
  { name: "Alice", age: 25, isStudent: false },
  { name: "Bob", age: 17, isStudent: true }
];

const double = (x: number): number => x * 2;

function createDefaultPerson(): Person {
  return {
      name: "Unknown",
      age: 0,
      isStudent: false
  };
}

File: lab_2/output/input3.py
@dataclass
class Person:
    name: str
    age: float
    isStudent: bool

@dataclass
class Book:
    title: str
    pages: float

def greet(name: str) -> None:
    print(f"Hello, {name}not ")

def add(x: float, y: float) -> float:
    result = x + y
    return result

def createPerson(name: str, age: float, isStudent: bool) -> Person:
    return {
    name: name,
    age: age,
    isStudent: isStudent
    }

def updateBook(book: Book, newTitle: str) -> Book:
    updated = { title: newTitle, pages: book.pages }
    print(f"Book updated to: {newTitle}")
    return updated

def isAdult(person: Person) -> bool:
    return person.age >= 18

numbers = [1, 2, 3, 4, 5]

users = [
  { name: "Alice", age: 25, isStudent: False },
  { name: "Bob", age: 17, isStudent: True }
]

double = lambda x: x * 2  # type: (float) -> float

def createDefaultPerson() -> Person:
    return {
    name: "Unknown",
    age: 0,
    isStudent: False
    }